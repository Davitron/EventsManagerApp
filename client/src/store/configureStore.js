import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
import rootReducer from '../reducers/rootReducer';

const configureStore = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk))
);

configureStore.subscribe(() => {
  console.log('STORE SUB...', configureStore.getState());
});

export default configureStore;
