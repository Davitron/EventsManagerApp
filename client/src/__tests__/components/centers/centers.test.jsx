import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import ConnectedCenter, { Center } from '../../../components/center/centers';

jest.mock('../../../helpers/image-upload', () => jest.fn(() => Promise.resolve({})));

let mountedComponent;

let props;
const locations = [];
const history = {
  push(location) {
    locations.push(location);
  },
};

const response = {
  centers: {},
  newCenter: {},
  allStates: {},
};

const dispatch = fn => fn;

const createCenter = jest.fn();

const location = {};


const mockStore = configureStore();
let wrapper;


/**
 * @description Initialise the component
 *
 * @returns {object} mountedComponent - Mounte
 */
const getComponent = () => {
  if (!mountedComponent) {
    props = {
      dispatch,
      response,
      history,
      location,
      createCenter
    };
    history.push = jest.fn();
    mountedComponent = shallow(<Center {...props} />);
  }
  return mountedComponent;
};

describe('Center Component', () => {
  beforeEach(() => {});

  it('component successfully rendered', () => {
    wrapper = getComponent();
    expect(wrapper).toMatchSnapshot();
  });

  it('testing component will recieve props', () => {
    const nextProps = {
      response: {
        ...response,
        centers: {
          status: 'failed',
          data: {
            message: 'Error when fetching',
            data: null
          }
        },
        newCenter: {
          status: 'success',
          data: {
            name: 'new center',
            address: 'a test adress for test'
          }
        }
      },
      location: {
        search: '/centers'
      }
    };

    wrapper = getComponent();

    const centersResponse = nextProps.response.centers;
    wrapper.instance().componentWillReceiveProps(nextProps);
    expect(wrapper.state().serverError).toEqual(centersResponse.data.message);
    expect(wrapper.state().openModal).toEqual(false);
    expect(wrapper.state().isRequestMade).toEqual(false);
  });

  it('testing component will recieve props', () => {
    const nextProps = {
      response: {
        ...response,
        centers: {
          status: 'success',
          data: {
            message: 'Centers Retrieved',
            data: [
              {
                id: 1,
                name: 'new center',
                address: 'a test adress for test'
              },
              {
                id: 2,
                name: 'new center2',
                address: 'a test adress for test in another place'
              }
            ],
            metadata: {
              pagination: {
                page: 1,
                total: 10,
                limit: 2,
                offset: 0
              }
            }
          }
        },
        newCenter: {
          status: 'success',
          data: {
            name: 'new center',
            address: 'a test adress for test'
          }
        },
        allStates: {
          status: 'success',
          data: [
            {
              id: 1,
              stateName: 'abia'
            },
            {
              id: 2,
              stateName: 'imo'
            }
          ]
        }
      },
      location: {
        search: '/centers'
      }
    };

    wrapper = getComponent();

    const centersResponse = nextProps.response.centers;
    wrapper.instance().componentWillReceiveProps(nextProps);
    expect(wrapper.state().serverError).toEqual(null);
    expect(wrapper.state().data).toEqual(centersResponse.data.data);
    expect(wrapper.state().loading).toEqual(false);
    expect(wrapper.state().pagingData).toEqual(centersResponse.data.metadata.pagination);
    expect(wrapper.state().states).toEqual(nextProps.response.allStates.data);
  });

  it('testing on page change function', () => {
    wrapper = getComponent();
    wrapper.instance().onChangePage(1);
    expect(history.push).toHaveBeenCalled();
  });

  it('testing on page size change function', () => {
    wrapper = getComponent();
    wrapper.instance().onPageSizeChange(1, 9);
    expect(history.push).toHaveBeenCalled();
  });

  it('testing on on search function', () => {
    wrapper = getComponent();
    wrapper.instance().onSearch({ page: 1, limit: 9 });
    expect(history.push).toHaveBeenCalled();
  });

  it('testing on onSubmit function', () => {
    wrapper = getComponent();
    const event = { preventDefault: () => undefined };
    event.preventDefault = jest.fn();
    wrapper.instance().onSubmit(event);
    expect(wrapper.state().isRequestMade).toEqual(false);
  });

  it('testing on onSubmit function', () => {
    wrapper = getComponent();
    const event = { preventDefault: () => undefined };
    event.preventDefault = jest.fn();
    wrapper.instance().onSubmit();
    expect(wrapper.state().isRequestMade).toEqual(false);
  });


  it('testing on onSubmit function', () => {
    const center = {
      name: 'Test Arena',
      stateId: 2,
      address: 'This is a test address for this event center',
      image: {
        files: []
      },
      hallCapacity: '500',
      carParkCapacity: '200',
      facilities: [
        'cctv',
        'media support',
        'projector'
      ],
      price: '100000'
    };
    wrapper = getComponent();
    wrapper.instance().onSubmit(center);
    expect(wrapper.state().openModal).toEqual(false);
    expect(wrapper.state().isRequestMade).toEqual(true);
  });

  it('testing on open modal function', () => {
    wrapper = getComponent();
    wrapper.instance().showModal();
    wrapper.setState({
      role: 'admin'
    });
    expect(wrapper.state().openModal).toEqual(true);
  });

  it('testing on hide modal function', () => {
    wrapper = getComponent();
    wrapper.instance().hideModal();
    expect(wrapper.state().openModal).toEqual(false);
  });
});

describe('Connected Component', () => {
  describe('Connected Centers Component', () => {
    it('component successfully rendered', () => {
      const store = mockStore({
        response,
      });
      wrapper = shallow(<ConnectedCenter store={store} history={history} location={location} />);
      expect(wrapper.length).toBe(1);
    });
  });
});
