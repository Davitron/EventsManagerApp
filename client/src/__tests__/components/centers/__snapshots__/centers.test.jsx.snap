// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Center Component component successfully rendered 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Center
    createCenter={[MockFunction]}
    dispatch={[Function]}
    getAll={[Function]}
    getStates={[Function]}
    history={
      Object {
        "push": [MockFunction],
      }
    }
    location={Object {}}
    response={
      Object {
        "allStates": Object {},
        "centers": Object {},
        "newCenter": Object {},
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="background"
      >
        <div
          className="my-container"
        >
          <div
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            <span
              style={
                Object {
                  "fontSize": "45px",
                }
              }
            >
              Centers
            </span>
          </div>
          <SearchForm
            onSearch={[Function]}
            states={Array []}
          />
          <Grid>
            <GridRow
              colunms={3}
            >
              
            </GridRow>
          </Grid>
        </div>
        <Connect(CenterFormModal)
          errors={Object {}}
          hideModal={[Function]}
          isRequestMade={false}
          onSubmit={[Function]}
          open={false}
          states={Array []}
        />
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="my-container"
          >
            <div
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <span
                style={
                  Object {
                    "fontSize": "45px",
                  }
                }
              >
                Centers
              </span>
            </div>
            <SearchForm
              onSearch={[Function]}
              states={Array []}
            />
            <Grid>
              <GridRow
                colunms={3}
              >
                
              </GridRow>
            </Grid>
          </div>,
          <Connect(CenterFormModal)
            errors={Object {}}
            hideModal={[Function]}
            isRequestMade={false}
            onSubmit={[Function]}
            open={false}
            states={Array []}
          />,
          false,
        ],
        "className": "background",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                style={
                  Object {
                    "textAlign": "center",
                  }
                }
              >
                <span
                  style={
                    Object {
                      "fontSize": "45px",
                    }
                  }
                >
                  Centers
                </span>
              </div>,
              <SearchForm
                onSearch={[Function]}
                states={Array []}
              />,
              <Grid>
                <GridRow
                  colunms={3}
                >
                  
                </GridRow>
              </Grid>,
              false,
            ],
            "className": "my-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span
                  style={
                    Object {
                      "fontSize": "45px",
                    }
                  }
                >
                  Centers
                </span>,
                "style": Object {
                  "textAlign": "center",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Centers",
                  "style": Object {
                    "fontSize": "45px",
                  },
                },
                "ref": null,
                "rendered": "Centers",
                "type": "span",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "onSearch": [Function],
                "states": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <GridRow
                  colunms={3}
                >
                  
                </GridRow>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    false,
                    "",
                    Array [],
                  ],
                  "colunms": 3,
                },
                "ref": null,
                "rendered": Array [
                  false,
                  "",
                ],
                "type": [Function],
              },
              "type": [Function],
            },
            false,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "errors": Object {},
            "hideModal": [Function],
            "isRequestMade": false,
            "onSubmit": [Function],
            "open": false,
            "states": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="background"
        >
          <div
            className="my-container"
          >
            <div
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <span
                style={
                  Object {
                    "fontSize": "45px",
                  }
                }
              >
                Centers
              </span>
            </div>
            <SearchForm
              onSearch={[Function]}
              states={Array []}
            />
            <Grid>
              <GridRow
                colunms={3}
              >
                
              </GridRow>
            </Grid>
          </div>
          <Connect(CenterFormModal)
            errors={Object {}}
            hideModal={[Function]}
            isRequestMade={false}
            onSubmit={[Function]}
            open={false}
            states={Array []}
          />
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="my-container"
            >
              <div
                style={
                  Object {
                    "textAlign": "center",
                  }
                }
              >
                <span
                  style={
                    Object {
                      "fontSize": "45px",
                    }
                  }
                >
                  Centers
                </span>
              </div>
              <SearchForm
                onSearch={[Function]}
                states={Array []}
              />
              <Grid>
                <GridRow
                  colunms={3}
                >
                  
                </GridRow>
              </Grid>
            </div>,
            <Connect(CenterFormModal)
              errors={Object {}}
              hideModal={[Function]}
              isRequestMade={false}
              onSubmit={[Function]}
              open={false}
              states={Array []}
            />,
            false,
          ],
          "className": "background",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  style={
                    Object {
                      "textAlign": "center",
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "fontSize": "45px",
                      }
                    }
                  >
                    Centers
                  </span>
                </div>,
                <SearchForm
                  onSearch={[Function]}
                  states={Array []}
                />,
                <Grid>
                  <GridRow
                    colunms={3}
                  >
                    
                  </GridRow>
                </Grid>,
                false,
              ],
              "className": "my-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span
                    style={
                      Object {
                        "fontSize": "45px",
                      }
                    }
                  >
                    Centers
                  </span>,
                  "style": Object {
                    "textAlign": "center",
                  },
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Centers",
                    "style": Object {
                      "fontSize": "45px",
                    },
                  },
                  "ref": null,
                  "rendered": "Centers",
                  "type": "span",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "onSearch": [Function],
                  "states": Array [],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <GridRow
                    colunms={3}
                  >
                    
                  </GridRow>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      false,
                      "",
                      Array [],
                    ],
                    "colunms": 3,
                  },
                  "ref": null,
                  "rendered": Array [
                    false,
                    "",
                  ],
                  "type": [Function],
                },
                "type": [Function],
              },
              false,
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "errors": Object {},
              "hideModal": [Function],
              "isRequestMade": false,
              "onSubmit": [Function],
              "open": false,
              "states": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          false,
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
